python/hpsmc/alignment/_apply.py

```diff
@@ -0,0 +1,139 @@
+"""! applying pede results to a detector in hps-mc jobs"""
+
+import shutil
+import re
+import os
+import logging
+
+from hpsmc.component import Component
+from ._parameter import Parameter
+
+class ApplyPedeRes(Component) :
+    """! Apply a millepede.res file to a detector description
+
+    This job component loads a result file into memory and the
+    goes line-by-line through a detector description, updating
+    the lines with any parameters that have updated values in the 
+    result file.
+    """
+
+    logger = logging.getLogger('ApplyPedeRes')
+
+    def __init__(self) :
+        # config
+        self.java_dir = None
+
+        # required job
+        self.detector = None
+
+        # optional job
+        self.res_file = 'millepede.res'
+        self.bump = True
+        self.force = False
+
+        super().__init__('ApplyPedeRes')
+
+    def required_config(self) :
+        return ['java_dir']
+
+    def required_parameters(self) :
+        return ['detector']
+
+    def optional_parameters(self) :
+        return ['res_file','bump','force']
+
+    def _detector_dir(self) :
+        return os.path.join(self.java_dir, 'detector-data', 'detectors', self.detector)
+
+    def cmd_line_str(self) :
+        return 'custom python execute'
+
+    def execute(self, log_out, log_err) :
+        if self.bump :
+            # deduce source directory and check that it exists
+            src_path = self._detector_dir()
+            if not os.path.isdir(src_path) :
+                ApplyPedeRes.logger.error(f'Detector {self.detector} is not in hps-java')
+                return 1
+            
+            # deduce iter value, using iter0 if there is no iter suffix
+            matches = re.search('.*iter([0-9]*)', self.detector)
+            if matches is None :
+                ApplyPedeRes.logger.error('No "_iterN" suffix on detector name.')
+                return 2
+            else :
+                i = int(matches.group(1))
+                self.detector = self.detector.replace(f'_iter{i}',f'_iter{i+1}')
+    
+            # deduce destination path, and make sure it does not exist
+            dest_path = self._detector_dir()
```

Comment:
From what I can tell, there are two options for the name used for the new detector. It will just replace the one you are starting with or bump will add an iterN suffix to the file name with the appropriate N. I would also like a way to specify the name of the new detector in the job.json file, so can you also add a parameter that will be used for the new detector name and use that if it is provided and if not do what you have currently?